### ISSUE TITLE
potentially add new syntax

### ISSUE THREAD

**Issue by houtanb** [DynareTeam/dseries#23]
Imagine a `dseries` declared as:
```
s = dseries(rand(3,6), 1, {'a1','a2','b1','b2','c1','c2'});
```
Currently, we can select all series that end with a `2` by typing:
```
s{'@a,b,c@2'}
```
It would be nice to have a wildcard syntax that would simplify this. Something like:
```
s{'*2'}
```
or
```
s{'@*@2'}
```

**Comment by stepan-a** [comment]
We already have the possibility to use regular expressions using square brackets. In this example:
```matlab
s{'[a-z]2'}
```
should do the selection you want. Using regular expression is more precise (for instance, if you had a variable called `to1to2` it would be also selected by a wildcard syntax)

**Comment by houtanb** [comment]
@stepan-a this is great. I didn't know this. Why not also support `*`?

**Comment by stepan-a** [comment]
Yes but he would also have `ANYTHING_WITH_LETTERS_NUMBERS_AND_OTHER_SYMBOLS_US` if the variable exists... The regular expression `[A-Za-z]` is not equivalent to `*`, I suppose that the regular expression you have in mind for the wildcard would be `[\w*]`. So, in principle we should just have to replace all the occurrences of `*` by `[\w*]` (if they are not surrounded by square brackets). My main concern is that the symbol `*` in a regular expression is not a wildcard parameter... So I think that this may be confusing. If we really want a wildcard parameter, maybe we should use another symbol (like `~`, but this need to be checked).

**Comment by houtanb** [comment]
@stepan-a what you say is exactly right and I would agree with the substitution of `*` with `[\w*]` because I would imagine it to be the most common case of a user wanting regular expression matching. It would further save the basic user from having to mess with regular  this `expressions, which can be time-consuming, even for this basic one if s/he doesn't have experience with them. I think that `*` is better than `~` because it already has meaning in bash and in the Matlab/Octave shell, so many users might even use it intuitively, without knowing about regular expressions.

**Comment by MichelJuillard** [comment]
Is [\w*] (with the brackets)  a correct regexp?

**Comment by houtanb** [comment]
@MichelJuillard you are right. It should be `\w*` or `[\w]*`, though I wonder if @stepan-a imposes the `[]` to make it easier to search for the regular expression. I am not sure where/whether the syntax is documented.

**Comment by MichelJuillard** [comment]
Thanks @houtanb. @stepan-a  I think there is a potential problem here:
 - [a-z] is a correct regexp
 - [\w*] isn't a correct regexp
You have to clarify the use of [ ]: either their usual meaning inside a regexp or a dseries syntax introducing a regexp

**Comment by houtanb** [comment]
So there is a question with the syntax. Given
```
>> a = dseries(rand(3,3),'1999y',{'GDP_US','GDP_FR','cpi_US'})
 
a is a dseries object:
 
      | GDP_US  | GDP_FR   | cpi_US 
1999Y | 0.35095 | 0.075967 | 0.18391
2000Y | 0.51325 | 0.23992  | 0.23995
2001Y | 0.40181 | 0.12332  | 0.41727
```
we have the following:
```
>> a{'[\w]*_US'}
 
ans is a dseries object:
 
      | GDP_US  | cpi_US 
1999Y | 0.35095 | 0.18391
2000Y | 0.51325 | 0.23995
2001Y | 0.40181 | 0.41727
 
>> a{'[\w*]_US'}
 
ans is a dseries object:
 
      | GDP_US  | cpi_US 
1999Y | 0.35095 | 0.18391
2000Y | 0.51325 | 0.23995
2001Y | 0.40181 | 0.41727
 
>> a{'\w*_US'}
Error using dseries/extract (line 61)
dseries::extract: Variable \w*_US is not a member of A!

Error in dseries/subsref (line 236)
        B = extract(A,S(1).subs{:});
```
@stepan-a perhaps this is an oversight. If not, if you need something to denote the beginning/end of a regular expression, can we use parenthesis instead of  brackets? Something like: `a{'(\w*)_US'}`, `a{'([A-Za-z]*)_US'}`, etc.

**Comment by stepan-a** [comment]
The brackets are used in a string to say that the user intends to use a regular expression, as explained  in the manual reference [here](https://dynare.adjemian.eu/doc-unstable/timeseries.html#C%20=%20extract) (I profit the occasion to show you the entirely new  implementation of the reference manual in Sphinx). Inside these square brackets, one can put any regular expression as described in the matlab documentation (in principle we could, for instance,  add other square brackets, but I need to check).

@houtanb In your example, `a{'[\w*]_US'}` is the correct syntax I had in mind. But I do not understand why `a{'[\w]*_US'}` works, it should not. There must be a bug I need to fix here.

Using parenthesis to delimit regular expression is not better... Since we can also use them to write regular expressions.

**Comment by houtanb** [comment]
@stepan-a there are more problems:
```
>> a = dseries(rand(3,3),'1999y',{'GDP_US','GDP_FR','cpi_US'})
 
a is a dseries object:
 
      | GDP_US   | GDP_FR  | cpi_US 
1999Y | 0.049654 | 0.49086 | 0.90005
2000Y | 0.90272  | 0.48925 | 0.36925
2001Y | 0.94479  | 0.33772 | 0.1112 
 
>> a{'[G{1}\w*]'}
 
ans is a dseries object:
 
      | GDP_US   | GDP_FR  | cpi_US 
1999Y | 0.049654 | 0.49086 | 0.90005
2000Y | 0.90272  | 0.48925 | 0.36925
2001Y | 0.94479  | 0.33772 | 0.1112 
```
The output of the second command should not contain the third column

**Comment by stepan-a** [comment]
I fixed the issues with regular expressions in the master branch. The correct syntax to select variables with regular expressions is to open and close a string with square brackets, for instance:
```matlab
o{'[\w*_US]'}
```
Without the square brackets the string is not interpreted as a regular expression (`subsref` and `extract` methods will probably return an error). I checked all the examples posted by @houtanb. Obviously other square brackets, parenthesis or curly braces can be used in the regular expression.

Also I added a wildcard parameter. I do not use `*` to avoid conflicts with regular expressions, and use `~` instead.
